spec/rspec/core/output_wrapper_spec.rb

```diff
@@ -0,0 +1,37 @@
+module RSpec::Core
+  RSpec.describe OutputWrapper do
+    let(:output) { double('IO') }
+    let(:wrapper) { OutputWrapper.new(output) }
+
+    describe '#output=' do
+      let(:another_output) { StringIO.new }
+
+      it 'changes the output stream' do
+        wrapper.output = another_output
+        expect(wrapper.output).to eq(another_output)
+      end
+    end
+
+    [:binmode, :binmode?, :bytes, :chars, :close, :close_on_exec=,
+     :close_on_exec?, :close_read, :close_write, :closed?, :codepoints, :each, :each_byte, :each_char,
+     :each_codepoint, :each_line, :eof, :eof?, :external_encoding, :fcntl, :fdatasync, :fileno, :flush,
+     :fsync, :getbyte, :getc, :gets, :inspect, :internal_encoding, :ioctl, :isatty, :lineno, :lineno=,
+     :lines, :pid, :pos, :pos=, :print, :printf, :putc, :puts, :read, :read_nonblock, :readbyte, :readchar,
+     :readline, :readlines, :readpartial, :reopen, :rewind, :seek, :set_encoding, :stat, :sync, :sync=,
+     :sysread, :sysseek, :syswrite, :tell, :to_i, :to_io, :tty?, :ungetbyte, :ungetc, :write, :write_nonblock].each do |method|
+      describe "##{method}" do
+        let(:block) { Proc.new { 'test block' } }
+
+        it "calls output's '#{method}' with the same params" do
+          expect(output).to receive(method).with('all', 'args')
+          wrapper.send(method, 'all', 'args')
+        end
+        it "calls output's '#{method}' with the same params and block" do
+          expect(output).to receive(method).with('all', 'args', &block)
+          wrapper.send(method, 'all', 'args', &block)
+        end
+      end
+    end
```

Comment:
I'm not sure that testing each method like this really does anything for us other than bloating our spec site with tons of specs.  Given they are defined in a loop I'm not too worried about a particular one not working properly. Plus this is going to be exercised constantly by all the other specs (anything involving formatter output!) that I'm not worried about it breaking on us.  

Instead, I think I'd like to see specs that demonstrate what this class is for: acting like an IO while allowing redirection to a different IO after instantiation.  To that end, I think you can write some a spec that instantiates it with a `StringIO` and calls IO methods to show that it writes to the underlying StringIO.  Then you can write another spec that shows that when you change the output it writes new output to that StringIO instance instead.